cli/npm/resolvers/common.rs

```diff
@@ -95,24 +97,47 @@ pub fn ensure_registry_read_permission(
   permissions: &dyn NodePermissions,
   registry_path: &Path,
   path: &Path,
+  registry_cache: &Mutex<HashMap<PathBuf, PathBuf>>,
+  path_cache: &Mutex<HashMap<PathBuf, PathBuf>>,
 ) -> Result<(), AnyError> {
   // allow reading if it's in the node_modules
   if path.starts_with(registry_path)
     && path
       .components()
       .all(|c| !matches!(c, std::path::Component::ParentDir))
   {
-    // todo(dsherret): cache this?
-    if let Ok(registry_path) = fs.realpath_sync(registry_path) {
-      match fs.realpath_sync(path) {
-        Ok(path) if path.starts_with(registry_path) => {
-          return Ok(());
+    let registry_path_canon =
+      match registry_cache.lock().unwrap().get(registry_path) {
```

Comment:
Since only one thread will ever be entering here, I believe it's ok to just acquire the lock once at the start of the function.